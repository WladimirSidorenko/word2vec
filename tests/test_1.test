#!/bin/sh

##################################################################
# Variables
BIN=${1}
INPUT_0='test_1.0.in'
INPUT_1='test_1.1.in'
OUTPUT_0='test_1.0.out'
OUTPUT_1='test_1.1.out'
EXPECTED_0='test_1.0.expected'
EXPECTED_1='test_1.1.expected'
TEST_NAME='task_specific'

##################################################################
# Test 0
echo '1..2'
${BIN} -min-count 0 -train "${INPUT_0}" -output "${OUTPUT_0}" -threads 1
if test $? -eq 0 && `diff -q "${OUTPUT_0}" "${EXPECTED_0}" > /dev/null`; then
    echo 'ok 1 # trained word vectors match the task'
else
    echo 'not ok 1 # trained word vectors do not match the tasks'
fi

${BIN} -min-count 0 -iter 500 -size 2 -ts 1 -threads 1 \
    -train "${INPUT_1}" -output "${OUTPUT_1}"
if test $? -eq 0 && `diff -q "${OUTPUT_1}" "${EXPECTED_1}" > /dev/null`; then
    echo 'ok 2 # trained word vectors match the task'
else
    echo 'not ok 2 # trained word vectors do not match the tasks'
fi
