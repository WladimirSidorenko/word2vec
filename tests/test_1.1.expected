6 2
</s> 0.068329 0.151054 
hello -1.842445 -0.050096 
world -2.337824 0.096693 
you -2.176676 -0.017533 
are -2.047771 -0.053567 
beautiful -2.867711 -0.081639 
